pkg/operator/starter.go

```diff
@@ -45,54 +50,89 @@ const (
 	resync = 20 * time.Minute
 )
 
-func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
-	kubeClient, err := kubeclient.NewForConfig(rest.AddUserAgent(controllerConfig.KubeConfig, targetName))
+func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext, guestKubeConfigFile string) error {
+	isHyperShift := guestKubeConfigFile != ""
+	// Kubeconfig received through service account or --kubeconfig arg
+	// is for the management cluster.
+	managementKubeConfig := controllerConfig.KubeConfig
+	managementKubeClient, err := kubeclient.NewForConfig(rest.AddUserAgent(managementKubeConfig, targetName))
```

Comment:
In my PR I prefix variables with either `controlPlane` or `guest`. We should agree on something and use everywhere...

Reply:
I'm replaced `management` / `mgmt` with `controlPlane`. But then you should use `controlPlane` prefix for everything on the control plane - like `controlPlaneKubeClient`, `controlPlaneInformersForNamespace` etc. :-)